plutus-tx/src/PlutusTx/IsData/Class.hs

```diff
@@ -92,6 +92,18 @@ instance UnsafeFromData Builtins.BuiltinByteString where
     {-# INLINABLE unsafeFromBuiltinData #-}
     unsafeFromBuiltinData = BI.unsafeDataAsB
 
+instance ToData Builtins.BuiltinString where
+    {-# INLINABLE toBuiltinData #-}
+    toBuiltinData = mkB . BI.encodeUtf8
```

Comment:
Mandating UTF-8 is unsatisfactory, but we need these instances since `BultinString` is used in `Anchor`.

Reply:
Hmmm. This is a pretty important constraint for the ledger. But they must have some similar constraint, since that string must be represented in the tx format as some bytestring. So we should be able to do the same thing... Let's ask what they do.

Reply:
We do not enforce utf8 encoding. ByteStrings are just ByteString for us.

So, the only time we should bother with this, if there will be desire to read this as text in Plutus. Then we can try figure out if we should do some sort of validation on it on the ledger side (which we could, but I know many people would be against it)

Reply:
Since we don't need to use String in Anchor, this is no longer a concern!